src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1307,7 +1307,7 @@ private async ValueTask RefreshAsync()
                     Model.Context.ServiceBroker,
                     Model.Context.Projects,
                     CancellationToken.None);
-                _packageList.UpdatePackageStatus(installedPackages.ToArray());
+                _packageList.UpdatePackageStatus(installedPackages.ToArray(), true);
```

Comment:
I recall this being a hot path for many events in PMUI. I know we're considering refactoring a lot of this code at some point which may reduce the frequency as well. 
Just wondering how confident you feel that this isn't overly clearing items from the cache? I'll look later as well.

Reply:
I added a bool to allow the caller of "RefreshAsync" to determine if the cache should be cleared. I'm only passing true from "ExecuteAction" which should cover install/uninstall/update.

Reply:
Awesome!